plugins/inputs/turbostat/turbostat.go

```diff
@@ -0,0 +1,166 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux && amd64
+
+package turbostat
+
+import (
+	"bufio"
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/jpillora/backoff"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/turbostat/parser"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Turbostat struct {
+	Command []string        `toml:"command"`
+	Log     telegraf.Logger `toml:"-"`
+
+	cancel context.CancelFunc
+}
+
+func (*Turbostat) Gather(telegraf.Accumulator) error {
+	return nil
+}
+
+func (*Turbostat) SampleConfig() string {
+	return sampleConfig
+}
+
+// Start starts the plugin.
+// Maintains a Turbostat process and parses metrics from stdout.
+func (t *Turbostat) Start(a telegraf.Accumulator) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	go func() {
+		retryDelay := &backoff.Backoff{
+			Min:    1 * time.Second,
+			Max:    30 * time.Second,
+			Factor: 2,
+			Jitter: false,
+		}
+		for {
+			t.Log.Info("Starting Turbostat")
+			startedAt := time.Now()
+			err := t.runTurbostat(ctx, a)
+			t.Log.Errorf("Turbostat encountered an error: %s", err)
+			// Exit the goroutine if the context was cancelled.
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			// If Turbostat ran for more than 30s, there likely was
+			// no major issue and we reset the backoff to the minimum.
+			if time.Since(startedAt) > 30*time.Second {
+				retryDelay.Reset()
+			}
+			delay := retryDelay.Duration()
+			t.Log.Infof("Restarting Turbostat in %.1fs", delay.Seconds())
+			// Wait until the delay elapses or the context is cancelled.
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(delay):
+			}
+		}
+	}()
+	return nil
+}
+
+// Stop stops the plugin.
+func (t *Turbostat) Stop() {
+	t.cancel()
+}
+
+func init() {
+	inputs.Add("turbostat", func() telegraf.Input {
+		return &Turbostat{}
+	})
+}
```

Comment:
This should be at the bottom of the file.

Reply:
Move to the bottom in https://github.com/influxdata/telegraf/pull/17368.